src/NuGet.Core/NuGet.Configuration/PackageNamespaces/SearchNode.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace NuGet.Configuration
+{
+    internal class SearchNode
+    {
+        public readonly SearchNode Parent;
+        public readonly Dictionary<char, SearchNode> Children;
+        public bool IsValueNode => !string.IsNullOrWhiteSpace(NamespaceId);
```

Comment:
If we can remove `Parent` property, then we can remove `IsValueNode` property also.

Reply:
About `Parent` we have another thread for it. But have to keep at least one of `IsValueNode` or `NamespaceId` to make it qualify as trie. Currently it's internal type such small details are not important. I'll do another iteration of clean up for perf optimization once everything is more clear.